viewing_party/party.py

```diff
@@ -0,0 +1,235 @@
+    
+    return final_movie_list_of_dictionaries
+
+def get_friends_unique_watched(user_data):
+    #first, let's get the list of movies the user has watched
+    users_movies = []
+    for index in range(len(user_data["watched"])):    
+        users_movies.append(user_data["watched"][index]["title"])
+
+    #next, let's get the set of movies the friends have watched
+    friends_movies = set()
+
+    friends = user_data["friends"]
+    for friend in range(len(friends)):
+        watched_list = friends[friend]["watched"]
+        for movies in watched_list:
+            for individual_title in movies:
+                #print(movies[individual_title])
+                friends_movies.add(movies[individual_title])
+                
+    #convert the users_movies to a set for comparison
+    users_movies = set(users_movies)
+    
+    #find the difference
+    movies_at_least_one_friend_watched = friends_movies - users_movies
+
+    #convert to a list of dictionaries to return
+    final_movie_list_of_dictionaries = []
+    for movie in movies_at_least_one_friend_watched:
+        final_movie_list_of_dictionaries.append({"title": movie}) 
+    
+    return final_movie_list_of_dictionaries
+
+def get_available_recs(user_data):
+
+    #Determine a list of recommended movies. 
+    # A movie should be added to this list if and only if
+    #The user has not watched it
+    #At least one of the user's friends has watched
+    #The "host" of the movie is a service that is in the user's "subscriptions"
+
+    user_watched_list = user_data["watched"]
+    user_subscriptions = user_data["subscriptions"]
+    
+    recommended_movies = []
+
+    friends = user_data["friends"]
+    for friend in range(len(friends)):
+        watched_list = friends[friend]["watched"]
+        for movie in watched_list:
+            if movie["host"] in user_subscriptions and movie not in recommended_movies:
+                if len(user_data['watched']) > 0:
+                    count = 0
+                    for title in user_data["watched"]:
+                        if movie['title'] is not title['title']:
+                            recommended_movies.append(movie)
+                        else:
+                            count += 1
+
+                        if count == len(recommended_movies):
+                            recommended_movies = []      
+                
+                else:
+                    recommended_movies.append(movie)
+    
+    
+    return recommended_movies
+
+def get_new_rec_by_genre(user_data):
+    #we want to get the user's most frequently watched genre (we have a function for this)
+
+    recommended_movies = []
+    friends_watched_movies = []
+    users_watched_movies = user_data["watched"]
+
+    #get user's most watched genre:
+    most_watched = get_most_watched_genre(user_data)
+
+    #get movies only the friends have watched:
+    friends = user_data["friends"]
+    for friend in range(len(friends)):
```

Comment:
I recommend using `for friend in friends`, and only using the range when an index is also needed.